test/inference/UniverseInferenceTest.java

```diff
@@ -0,0 +1,38 @@
+package universe;
```

Comment:
Should these be moved to `src/test/java...` or some other location to avoid the test/tests directory names?